File: src/java/main/commands/RichJson_this.java
package commands;

import core.RichJsonContext;
import core.RichJsonParser;

public class RichJson_this extends RichJson_ref {

    @Override
    public Object execute(RichJsonParser parser, RichJsonContext context) throws Exception {
        if (context.currentMember == null || context.currentMember.toString().isEmpty() || context.currentMember.equals("this")) {
            return context.current;
        }

        return resolvePathFrom(context.current, parser, context);
    }
}

File: src/c#/main/commands/RichJson_this.cs
using System;
using main.core;

namespace main.commands
{
    public class RichJson_this : RichJson_ref
    {
        public new object Execute(RichJsonParser parser, RichJsonContext context)
        {
            if (context.CurrentMember == null || string.IsNullOrEmpty(context.CurrentMember.ToString()) || context.CurrentMember.Equals("this"))
            {
                return context.Current;
            }

            return ResolvePathFrom(context.Current, parser, context);
        }
    }
}